sphinx_needs/needs.py

```diff
@@ -356,6 +362,63 @@ def process_caller(app: Sphinx, doctree: nodes.document, fromdocname: str) -> No
     return process_caller
 
 
+def load_config_from_toml(app: Sphinx, config: Config) -> None:
+    """
+    Load config from toml file, if defined in conf.py
+    """
+    needs_config = NeedsSphinxConfig(config)
+    if needs_config.from_toml is None:
+        return
+    if isinstance(needs_config.from_toml, str):
+        toml_file = Path(needs_config.from_toml)
+        toml_path = ["needs"]
+    else:
+        try:
+            _toml_file, toml_path = needs_config.from_toml
+            assert isinstance(_toml_file, str), "First element must be a string"
+            assert isinstance(toml_path, (list, tuple)), "Second element must be a list"
+            toml_file = Path(_toml_file)
+        except Exception:
+            log_warning(
+                LOGGER,
+                "'needs_from_toml' is not a str or (str, list[str])",
+                "config",
+                None,
+            )
+            return
+
+    # resolve relative to confdir
+    toml_file = Path(app.confdir, toml_file).resolve()
+    if not toml_file.exists():
+        log_warning(
+            LOGGER,
+            f"'needs_from_toml' file does not exist: {toml_file}",
+            "config",
+            None,
+        )
+        return
+    try:
+        with toml_file.open("rb") as f:
+            toml_data = tomllib.load(f)
+        for key in toml_path:
+            toml_data = toml_data[key]
+        assert isinstance(toml_data, dict), "Data must be a dict"
```

Comment:
Can this also be a list like need types?

Reply:
In what use case would you want it to be a list?

Reply:
Nah I misunderstood the code at first.